src/overlays/ovl_i5/fox_ti.c

```diff
@@ -1,138 +1,5375 @@
+                    actor->vel.y = 5.0f;
+                    actor->vel.z = spC8.z;
+                    actor->gravity = 0.8f;
+                } else {
+                    actor->fwork[0] = (Rand_ZeroOne() * 10.0f) + 10.0f;
+                    actor->fwork[1] = (Rand_ZeroOne() * 10.0f) + 10.0f;
+                    actor->obj.pos.x = D_i5_801BBF00[i].unk_00.pos.x + boss->obj.pos.x;
+                    actor->obj.pos.y = D_i5_801BBF00[i].unk_00.pos.y + boss->obj.pos.y;
+                    actor->obj.pos.z = D_i5_801BBF00[i].unk_00.pos.z + boss->obj.pos.z;
+                    actor->obj.rot.x = D_i5_801BBF00[i].unk_00.rot.x + boss->obj.rot.x;
+                    actor->obj.rot.y = D_i5_801BBF00[i].unk_00.rot.y + boss->obj.rot.y;
+                    actor->obj.rot.z = D_i5_801BBF00[i].unk_00.rot.z + boss->obj.rot.z;
+                    Matrix_Push(&gCalcMatrix);
+                    Matrix_RotateY(gCalcMatrix, Rand_ZeroOne() * 360.0f * M_DTOR, 0);
+                    Matrix_MultVec3f(gCalcMatrix, &D_i5_801B8D0C, &spC8);
+                    Matrix_Pop(&gCalcMatrix);
+                    actor->vel.x = spC8.x;
+                    actor->vel.y = (f32) ((Rand_ZeroOne() < 0.5f) ? -1 : 1) * ((Rand_ZeroOne() * 2.0f) + 15.0f);
+                    actor->vel.z = spC8.z;
+                    actor->gravity = 2.5f;
+                }
+            }
+            D_i5_801BBF00[i].unk_26 = 4;
+            Audio_PlaySfx(0x2903B009, boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+        }
+    }
+    if (boss->fwork[38] != 0.0f) {
+        Math_SmoothStepToF(&D_i5_801BBEF4[15], D_i5_801BBEF4[24], 1.0f, D_i5_801BBEF4[32], 0.01f);
+        Math_SmoothStepToAngle(&D_i5_801BBEF4[16], D_i5_801BBEF4[25], 1.0f, D_i5_801BBEF4[33], 0.01f);
+    }
+    if (boss->fwork[39] != 0.0f) {
+        Math_SmoothStepToAngle(&D_i5_801BBEF4[18], D_i5_801BBEF4[27], 1.0f, D_i5_801BBEF4[34], 0.01f);
+        Math_SmoothStepToAngle(&D_i5_801BBEF4[19], D_i5_801BBEF4[28], 1.0f, D_i5_801BBEF4[34], 0.01f);
+    }
+    if (boss->fwork[0x28] != 0.0f) {
+        Math_SmoothStepToAngle(&D_i5_801BBEF4[21], D_i5_801BBEF4[30], 1.0f, D_i5_801BBEF4[35], 0.01f);
+        Math_SmoothStepToAngle(&D_i5_801BBEF4[22], D_i5_801BBEF4[31], 1.0f, D_i5_801BBEF4[35], 0.01f);
+    }
+    Math_SmoothStepToF(&boss->fwork[38], boss->fwork[35], 1.0f, 0.3f, 0.01f);
+    Math_SmoothStepToF(&boss->fwork[39], boss->fwork[0x24], 0.2f, 0.5f, 0.01f);
+    Math_SmoothStepToF(&boss->fwork[0x28], boss->fwork[37], 0.2f, 0.5f, 0.01f);
+    if (D_i5_801BBEF0[6] > 0) {
+        D_i5_801BBEF0[6]--;
+        if (D_i5_801BBEF0[6] == 0) {
+            for (i = 0; i < ARRAY_COUNTU(D_i5_801BBF00); i++) {
+                D_i5_801BBF00[i].unk_18 = D_i5_801BBF00[i].unk_1C = D_i5_801BBF00[i].unk_20 = 0.0f;
+            }
+        }
+    }
+
+    for (i = 0; i < 4; i++) {
+        if (boss->swork[i + 5] > 0) {
+            boss->swork[i + 5]--;
+        }
+    }
+
+    for (i = 0; i < 0x22; i++) {
```

Comment:
```suggestion
    for (i = 0; i < 34; i++) {
```
or perhaps array count